src/System.Linq/src/System/Linq/Check.cs

```diff
@@ -0,0 +1,20 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Linq
+{
+    [Flags]
+    internal enum Check
```

Comment:
I admit the name here is kind of ambiguous... suggestions on what to rename this would be very welcome. I'm currently debating between `Typecast` and `TypecastOptions`.